docs/DeveloperGuide.md

```diff
@@ -154,6 +154,46 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add/Deduct reward points feature
+
+#### Implementation
+
+From `Points` class, a new feature of adding and deducting from the points a customer has, was required.
+A cumulative points system also had to be introduced. Adding points should increase the cumulative points a customer
+has while deducting points should not affect the cumulative points of the customer.
+
+As a result, a new attribute, `cumulative` was introduced to `Points`.
+For the parsing of the `addpoints` command, I had to consider between different design choices.
+The one I ultimately settled on was merely utilising `Integer.valueOf` method to parse the points inputted by the user.
+Also, in the execution of `AddPointsCommand` class, the points to add would exist as an Integer object instead of a
+`Points` object. The addition or subtraction of points from the `Points` class would then be as simple as adding the
+points to current points and cumulative points when it is positive, and subtracting from current points when it is 
+negative. This simple design would be sufficient to achieve the intended feature.
+Lastly, to store the cumulative points of `Customer`, a new attribute, `cumulativePoints` was introduced to the
+`JsonAdaptedCustomer`.
+
+
+There were other designs I was considering, listed below.
```

Comment:
Maybe you can add another header here, e.g. Alternatives Considered

Reply:
apologies just saw this, added